torchmultimodal/utils/attention.py

```diff
@@ -38,3 +41,34 @@ def get_extended_attention_mask(attention_mask: Tensor) -> Tensor:
     )  # fp16 compatibility
 
     return extended_attention_mask
+
+
+def get_extended_attention_mask_for_decoder(attention_mask: Tensor) -> Tensor:
```

Comment:
Can you add some more detail on what this function is doing? It seems quite a bit more complicated than get_extended_attention_mask, which is really just broadcasting the existing mask to extra dimensions